pkg/reconciler/v1alpha1/service/service.go

```diff
@@ -213,20 +214,64 @@ func (c *Reconciler) reconcile(ctx context.Context, service *v1alpha1.Service) e
 	// Apply additional logic, once the generic data has been propagated.
 	switch {
 	case service.Spec.RunLatest != nil:
-		runLatestCheckTrafficMigrated(ss, &route.Status, &config.Status)
+		c.runLatestCheckTrafficMigrated(service, &route.Status, &config.Status)
+	case service.Spec.Release != nil:
+		c.releaseCheckTrafficMigrated(service, &route.Status, &config.Status)
 	}
 	service.Status.ObservedGeneration = service.Generation
 
 	return nil
 }
 
-func runLatestCheckTrafficMigrated(ss *v1alpha1.ServiceStatus, rs *v1alpha1.RouteStatus, cs *v1alpha1.ConfigurationStatus) {
+func (c *Reconciler) releaseCheckTrafficMigrated(s *v1alpha1.Service, rs *v1alpha1.RouteStatus, cs *v1alpha1.ConfigurationStatus) {
+	// If the service's RouteReady condition is in a happy state.
+	if rc := s.Status.GetCondition(v1alpha1.ServiceConditionRoutesReady); rc != nil && rc.Status == corev1.ConditionTrue {
+		// Case 1. Either all the traffic is going to the `current` or there is no `candidate` at all.
+		// Note as implemented, if we have 2 revisions in the Spec, but route is ready to serve
+		// only `current`, but `rolloutPercent` is 0, then we'll presume that the service is ready,
+		// since it is _ready_ to serve the desired configuration.
+		rp := s.Spec.Release.RolloutPercent
+		revs := s.Spec.Release.Revisions
+		if rp == 0 || len(revs) == 1 {
+			if len(rs.Traffic) == 0 || rs.Traffic[0].RevisionName != revs[0] {
+				c.Logger.Debugf("%s: traffic not yet migrated to %s", s.Name, revs[0])
+				s.Status.MarkRouteNotYetReady()
+			}
```

Comment:
I feel like there's an uncovered path through here when I go from:
```yaml
revisions: ["rev1", "rev2"]
percent: 0
```

to:
```yaml
revisions: ["rev1", "rev3"]
percent: 0
```

We only check that "current" matches, not that "candidate" matches.

Reply:
Well, that's why I left the comment. Since this does not affect the traffic — should it matter?
If you think it does, then it's just reorganization of code.